action/protocol/rewarding/reward.go

```diff
@@ -222,7 +222,13 @@ func (p *Protocol) GrantEpochReward(
 	}
 
 	// Reward additional bootstrap bonus
-	if epochNum <= a.foundationBonusLastEpoch || (epochNum >= p.foundationBonusP2StartEpoch && epochNum <= p.foundationBonusP2EndEpoch) {
+	var grantFB bool
+	if a.hasFoundationBonusExtension() {
+		grantFB = a.grantFoundationBonus(epochNum)
+	} else {
+		grantFB = epochNum <= a.foundationBonusLastEpoch || (epochNum >= p.cfg.FoundationBonusP2StartEpoch && epochNum <= p.cfg.FoundationBonusP2EndEpoch)
```

Comment:
grant bonus depends on both `admin{}` stored in statedb, and `P2Start/End` in local struct

at Kamchatka height, we add the bonus Start/End epoch, so it solely depends on `admin{}` stored in statedb